test/src/server/routes/editor.js

```diff
@@ -35,29 +36,83 @@ describe('getEditorActivity', () => {
 	});
 	after(truncateEntities);
 
-	it('should return values for plotting graph', async () => {
+	it('should return revision data for graph', async () => {
 		const startDate = new Date();
-		startDate.setFullYear(2019, 1, 1);
-		const ActivityData = await getEditorActivity(editorJSON.id, startDate, Revision);
-		const MonthArray = ['Jan-19', 'Feb-19', 'Mar-19', 'Apr-19', 'May-19', 'Apr-19', 'May-19',
-			'Jun-19', 'Aug-19', 'Sep-19', 'Oct-19', 'Nov-19', 'Dec-19'];
+		startDate.setFullYear(2019, 0, 1);
+		const activityData = await getEditorActivity(editorJSON.id, startDate, Revision);
 
-		for (let i = 0; i < 12; i++) {
-			expect(ActivityData[MonthArray[i]]).to.equal(1);
-		}
+		// The result we expect from the function
+		const expectedResults = {
+			// eslint-disable-next-line sort-keys
+			'Jan-19': 1, 'Feb-19': 1, 'Mar-19': 1, 'Apr-19': 1, 'May-19': 1, 'Jun-19': 1, 'Jul-19': 1,
+			// eslint-disable-next-line sort-keys
+			'Aug-19': 1, 'Sep-19': 1, 'Oct-19': 1, 'Nov-19': 1, 'Dec-19': 1
+		};
+
+		// Adds remaining months upto the present month
+		const expectedStartDate = new Date();
+		expectedStartDate.setFullYear(2020, 0, 1);
+		const monthsUptoNow = eachMonthOfInterval({
```

Comment:
Don't do like this. This will always give correct result because this is exactly the way it's been fetched in the `getEditorActivity` function.
Rather create an `expectedResult` object with hard coded values

Reply:
The problem with creating hardcoded values is that our `getEditorActivity` function returns values till the present date.

So suppose if I do hardcode it till `Apr-20` then next month we'll need to add `May-20` so that the test does not fail next month. I personally don't think hard coding the values would be a good option. 

What else we can do here is pass an `endDate` into the `getEditorActivity` function default to the `Date.now()`. Then I think we can hardcode here with a precise end date. What do you think?

Reply:
Oh yeah I didn't think of this. Nice catch!
Yeah I think passing an endDate would better.

Reply:
I agree, passing and endDate is the way to go, not only just for the purpose of testing but also for more granular use ("fetch my activity from january of this year")